torch_xla/distributed/zero_redundancy_optimizer.py

```diff
@@ -411,25 +440,31 @@ def allgather_weights_and_update_full_parameter(self):
         if param.grad is not None or (self.use_grad_acc_hook and
                                       hasattr(shard, 'main_grad')):
           shard_data = shard.data
-          shard_data = shard_data.to(dtype=param.dtype)
+          if not self.higher_cc_precision:
```

Comment:
now we always use all gather with dtype==param.dtype, do not change this line

Reply:
Done